packages/cap-config-writer/src/cap-writer/package-json.ts

```diff
@@ -0,0 +1,153 @@
+import type { Package } from '@sap-ux/project-access';
+import { getCapCustomPaths } from '@sap-ux/project-access';
+import type { Editor } from 'mem-fs-editor';
+import path, { join } from 'path';
+import type { CdsUi5PluginInfo, CapServiceCdsInfo } from '../cap-config/types';
+import { enableCdsUi5Plugin, checkCdsUi5PluginEnabled, satisfiesMinCdsVersion, minCdsVersion } from '../cap-config';
+import type { Logger } from '@sap-ux/logger';
+import { t } from '../i18n';
+
+/**
+ * Retrieves the CDS watch script for the CAP app.
+ *
+ * @param {string} projectName - The name of the project.
+ * @param {string} appId - The ID of the app.
+ * @param {boolean} [useNPMWorkspaces] - Whether to use npm workspaces.
+ * @returns {{ [x: string]: string }} The CDS watch script for the CAP app.
+ */
+export function getCDSWatchScript(
+    projectName: string,
+    appId: string,
+    useNPMWorkspaces: boolean = false
```

Comment:
Can you update the doc to explain what is `projectName` and `appId`.  The current doc doesn't give any additional information. Does `projectName` mean cds project or ui5 application. Does `appId` include namespace or is this something else? 

Also we do not need the `useNPMWorkspaces` param if when calling you only pass it if its a workspace right?  The `appId` is only used for workspaces so instead:

```suggestion
    getCDSWatchScript(projectName, appid?)
```

Reply:
@IainSAP 
Updated the documentation for `projectName` and `appId`. Regarding `useNPMWorkspaces`, isn't `appId` always present regardless of the workspace configuration? Doesn't the watch script change based on the workspace setup, with the `--livereload false `option in the scripts? 
If `appId` is consistently available, live reload will remain disabled regardless of whether workspaces are enabled or disabled.

Reply:
as discussed I have made `appId` optional